network/default.go

```diff
@@ -1045,7 +1048,54 @@ func (n *network) processNetChan(listener tunnel.Listener) {
 
 				// add all the routes we have received in the sync message
 				for _, pbRoute := range pbNetSync.Routes {
+					// unmarshal the routes received from remote peer
 					route := pbUtil.ProtoToRoute(pbRoute)
+					// continue if we are the originator of the route
+					if route.Router == n.router.Options().Id {
+						log.Debugf("Network node %s skipping route addition: route already present", n.id)
+						continue
+					}
+					// lookup best routes for the services in the just received route
+					q := []router.QueryOption{
+						router.QueryService(route.Service),
+						router.QueryStrategy(n.router.Options().Advertise),
+					}
+
+					routes, err := n.router.Table().Query(q...)
+					if err != nil {
+						log.Debugf("Network node %s failed listing best routes for %s: %v", n.id, route.Service, err)
+						continue
+					}
+
+					// we found no routes for the given service
+					// create the new route we have just received
+					if len(routes) == 0 {
+						// add routes to the routing table
+						if err := n.router.Table().Create(route); err != nil && err != router.ErrDuplicateRoute {
+							log.Debugf("Network node %s failed to add route: %v", n.id, err)
+							continue
```

Comment:
This continue statement not needed if the above was added

Reply:
literally removed it right after you left